src/sestest/ListCertificatesCommand.cs

```diff
@@ -31,26 +31,99 @@ public Task<int> Execute(ListCertificatesCommandLine commandLine)
         {
             var now = DateTime.Now;
 
-            if (commandLine.ShowExpired)
+            var showCerts = TryParseShow(commandLine.Show);
+            if (!showCerts.HasValue)
             {
-                Logger.LogInformation("Including expired certificates.");
+                Logger.LogErrors(showCerts.Errors);
+                return Task.FromResult(-1);
             }
 
             var certificateStore = new CertificateStore();
             var allCertificates = certificateStore.FindAll();
-            var query = allCertificates.Where(c => c.HasPrivateKey)
-                .Where(c => now < c.NotAfter || commandLine.ShowExpired)
+            var query = allCertificates.Where(c => c.HasPrivateKey && c.RawData != null)
```

Comment:
nitpick: `query` sounds to me as if the collection is not enumerated yet, and `allCertificates` is unused after this point. Wondering about something like
```
var candidates = certificateStore.FindAll()
    .Where(c => c.HasPrivateKey && c.RawData != null)
    .ToList();
```

Reply:
Refactored after F2F discussion.